src/App.tsx

```diff
@@ -1,61 +1,173 @@
 import './App.scss';
+import { ChangeEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+};
+
+export type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
 
 export const App = () => {
+  const [todosList, setTodosList] = useState<(Todo & { user: User })[]>(() =>
+    todosFromServer.map(todo => {
+      const user = usersFromServer.find(u => u.id === todo.userId);
+
+      if (!user) {
+        alert('User not found');
+
+        return {
+          ...todo,
+          user: {
+            id: 0,
+            name: '',
+            username: '',
+            email: '',
+          },
+        };
+      }
+
+      return {
+        ...todo,
+        user,
+      };
+    }),
+  );
+
+  const [textForNewTodo, setTextForNewTodo] = useState<string>('');
+
+  const [selectedUser, setSelectedUser] = useState<string>('0');
+
+  const [hasErrorOnTitle, setHasErrorOnTitle] = useState<boolean>(true);
+
+  const [hasErrorOnUser, setHasErrorOnUser] = useState<boolean>(true);
+
+  function handleTextChange(e: ChangeEvent<HTMLInputElement>): void {
+    if (e.target.value !== '') {
+      setHasErrorOnTitle(false);
+    }
+
+    setTextForNewTodo(e.target.value);
+  }
+
+  function handleUserChange(e: ChangeEvent<HTMLSelectElement>): void {
+    if (e.target.value !== '0') {
+      setHasErrorOnUser(false);
+    }
+
+    setSelectedUser(e.target.value);
+  }
+
+  function handleCreateNewTodo(): void {
+    if (textForNewTodo !== '') {
+      setHasErrorOnTitle(false);
+    }
+
+    if (selectedUser !== '0') {
+      setHasErrorOnUser(false);
+    }
```

Comment:
The checks for `textForNewTodo` and `selectedUser` in `handleCreateNewTodo` are redundant since they are repeated later in the function. Consider removing these initial checks and relying on the subsequent checks that set the error states and return if the inputs are invalid.